File: streaming/examples/treasury/reducer.py
#!/usr/bin/env python

import sys
import os
sys.path.append(".")

try:
    from pymongo_hadoop import BSONReducer
    import pymongo_hadoop
except:
    here = os.path.abspath(__file__)
    module_dir = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(here))),
                    'language_support',
                    'python')
    sys.path.append(module_dir)
    from pymongo_hadoop import BSONReducer

def reducer(key, values):
    print >> sys.stderr, "Processing Key: %s" % key
    _count = _sum = 0
    for v in values:
        _count += 1
        _sum += v['bc10Year']
    return {'_id': key, 'avg': _sum / _count,
            'count': _count, 'sum': _sum }

BSONReducer(reducer)

File: streaming/examples/treasury/reducer.rb
#!/usr/bin/env ruby
require 'mongo-hadoop'

MongoHadoop.reduce do |key, values|
  count = sum = 0
  
  values.each do |value|
    count += 1
    sum += value['bc10Year']
  end

  { :_id => key, :average => sum / count }
end
